pom.xml

```diff
@@ -150,11 +150,15 @@
   </modules>
 
   <profiles>
-    <!-- A global profile defined for all modules for release-level verification. 
-      Optional processes such as building source and javadoc should be limited 
-      to this profile. -->
+
+    <!-- A global profile defined for all modules that enables a higher level of validation.
+      This profile is off by default to streamline development, but it is automatically enabled
+      during releases and precommit testing. Developers can enable it manually on the command line
+      using -Pall-checks.
+
+      Optional processes such as building source and javadoc should be limited to this profile. -->
     <profile>
-      <id>release</id>
+      <id>all-checks</id>
```

Comment:
It feels to me almost like there should be two separate profiles - all-checks and all-(tasks? builds?). all-tasks could be a superset of all-checks, or they could be disjoint.